modules/polaris/src/main/ts/ephox/polaris/words/Words.ts

```diff
@@ -26,9 +26,19 @@ const findUrlEnd = (characters: string[], startIndex: number): number => {
   return peakedWord.substr(0, 3) === '://' ? endIndex : startIndex;
 };
 
-const findWords = <T>(chars: T[], sChars: string[], characterMap: CharacterMap, options: WordOptions): T[][] => {
-  const words: T[][] = [];
-  let word: T[] = [];
+export type Word<T> = T[];
+interface WordIndex {
+  start: number;
+  end: number;
```

Comment:
I always forget this is necessary too.

```suggestion
  readonly start: number;
  readonly end: number;
```